web/src/server_events_dispatch.js

```diff
@@ -675,6 +677,19 @@ export function dispatch_normal_event(event) {
                 $("body").toggleClass("less_dense_mode");
                 $("body").toggleClass("more_dense_mode");
             }
+            if (event.property === "web_mark_read_on_scroll_policy") {
+                unread_ui.update_unread_banner();
+                if (
+                    message_lists.current.can_mark_messages_read_without_setting() &&
+                    !(
+                        event.value ===
+                        settings_config.web_mark_read_on_scroll_policy_values.never.code
+                    )
+                ) {
+                    message_lists.current.resume_reading();
+                    unread_ui.hide_unread_banner();
+                }
```

Comment:
This is a lot more business logic than we usually like to put directly in `server_events_dispatch`; can `unread_ui.update_unread_banner` do this?

Reply:
That's a good idea, since the logic between the two was similar I was able to shorten the code down and removed all this logic here.